datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -460,26 +436,17 @@ fn pop_expr(new_expr: &mut Vec<Vec<Expr>>) -> Result<Vec<Expr>> {
         .ok_or_else(|| DataFusionError::Internal("Failed to pop expression".to_string()))
 }
 
-fn to_arrays(
+fn populate_expr_set(
```

Comment:
maybe we could even make this a method on `ExprSet` to keep it more encapsulated

```rust
impl ExprSet {
  fn populate(
    &mut self,
    expr: &[Expr],
    input_schema: DFSchemaRef,
    expr_mask: ExprMask,
  )  -> Result<()> { 
  ...
  }
}
```